### ISSUE TITLE
update afternoon break episode 15 reflections 

### ISSUE THREAD

**Issue by alee** [swcarpentry/python-novice-gapminder#562]
Currently the episode 15 reflections text expects to happen after we go over functions but due to lesson episode reordering this is no longer the case. We either need to come up with other reflections or move the break after functions (which might be too long)

https://swcarpentry.github.io/python-novice-gapminder/15-coffee/index.html

https://swcarpentry.github.io/python-novice-gapminder/16-writing-functions/index.html

**Comment by DianahOnyino** [comment]
I concur with @alee. 

The requirements of reflection exersice during the afternoon coffee break are dependent on the students having knowledge of functions which comes in the next episode. I think it would be a good idea to move the reflection exercise after the functions episode to allow the students have a rich reflection exercise.

**Comment by knitwear25** [comment]
The reflection could be used as a pre-lab where you wish to challenge the attendees to think of what functions would help do even before starting the episode on "writing functions". This would help in two ways:
1. correct a wrong mental model that the student had
2. Affirm the concept when the instructor begins to discuss functions in the next episode

**Comment by grypy** [comment]
The reflection could be used as a pre-lab where you wish to challenge the attendees to think of what functions would help do even before starting the episode on "writing functions". This would help in two ways:

1. Correct a wrong mental model that the workshop attendee has.
2. Affirm the concept when the instructor begins to discuss functions in the next episode.

**Comment by experiment4life** [comment]
I attended this course last week,and liked the content of the three listed items.

However, "Reflection Exercise" as a title some how feels like we are reflecting upon what we have just learned, but this is relating to the following lesson. 

So we could change the title to "Preview Discussion" or something similar, which suggests that we will be learning about this in the next session. 

Also it would be nice to have these three discussion items answered somewhere in the upcoming lesson. 
Remember DRY which we talked about at the start of this lesson, here is a good example of that right here (which ever example we can point to). 
We cover Variable Scope later under "Local and Global Variable Use" tracking, so maybe we could add in one paragraph about the pros and cons of local vs global variables in there.

**Comment by bdemirb** [comment]
In workshops where learning materials are taught in multiple days, attendees can be asked to spend some time on reflection exercise after the end of each day. I think this would keep attendees motivated and give attendees a chance to learn deeper what they were taught in the workshop that day. There may not be enough time for attendees to focus on reflections exercises during coffee break as it may be not easy to digest all the new things they just learned.

For example, if attendees think of the exercise "What are the pros / cons of making a variable global or local to a function?" and come up with some ideas, suggestions, this would be really helpful for them in understanding the distinction between global and local variables. We tend to name variables in a meaningful way, however, if it is a local variable, we will not even see it during calling a function. Therefore, knowing whether a variable is locally used or globally defined will make attendees more engaged in the following lectures.

**Comment by JostMigenda** [comment]
I taught this course earlier this month and the “Reflection Exercise” that didn’t match what I’d just taught really tripped me up, so I’d be keen to deal with this issue.

Like @alee, I think it would not work to move the break to be after the Writing Functions episode; that would cram too much content between the lunch and coffee break, with relatively little content remaining after the coffee break. Therefore, my suggestion would be to change the exercise to ask learners to reflect on material covered before the coffee break.

For the episodes on `for` loops and looping over data sets, I’d suggest the following questions:
* Think about data you are working with. Is it split up into multiple files? Would you be able to process these files using a for loop?
* If you are creating a new data set, how can file names you choose make it easier or harder to loop over the data files?

An additional prompt on Conditionals would be useful, but I’m blanking on good ones right now. Does anyone have suggestions?
